src/services/store-service.js

```diff
@@ -1,43 +1,37 @@
 (function() {
     'use strict';
 
-    angular.module('openfin.store', ['angular-storage'])
-        .factory('storeService', ['store', '$rootScope', function(store, $rootScope) {
-            var initialStocks = {
-                'AAPL': 0,
-                'MSFT': 1,
-                'TITN': 2,
-                'SNDK': 3,
-                'TSLA': 4
-            };
-
-            var favouriteStocks = store.get('stocks') || initialStocks;
-
-            function order(obj) {
-                var tuples = [];
-
-                for (var key in obj) {
-                    if (obj.hasOwnProperty(key)) {
-                        tuples.push([key, obj[key]]);
-                    }
+    angular.module('openfin.store', ['angular-storage', 'openfin.desktop'])
+        .factory('storeService', ['store', 'desktopService', '$rootScope', function(store, desktopService, $rootScope) {
+            var KEY_NAME = 'windows';
+            var initialStocks = [
+                {
+                    id: 'OpenFinD3FC',
+                    stocks: [
+                        'AAPL', 'MSFT', 'TITN', 'SNDK', 'TSLA'
+                    ],
+                    closed: false
                 }
+            ];
+
+            var storage = JSON.parse(store.get(KEY_NAME)) || initialStocks;
 
-                tuples = tuples.sort(function(a, b) {
-                    return a[1] - b[1];
-                });
+            function getWindowStore() {
+                var windowIndex = storage.map(function(window) {
+                    return window.id;
+                }).indexOf(desktopService.getCurrentWindow().name);
 
-                return tuples.map(function(x) {
-                    return x[0];
-                });
+                return windowIndex > -1 ? storage[windowIndex] : undefined; // TODO: undefined?
             }
 
             function save(stock) {
-                store.set('stocks', favouriteStocks);
+                store.set(KEY_NAME, JSON.stringify(storage));
```

Comment:
Ideally this would only change the object for the current window, but what happens when another window saves? Would it overwrite the other window's changes, since the `initialStocks` variable is cached on creation, and so wouldn't reflect the updated values?